homeassistant/components/nice_go/coordinator.py

```diff
@@ -178,16 +182,24 @@ async def _update_refresh_token(self) -> None:
 
     async def client_listen(self) -> None:
         """Listen to the websocket for updates."""
-        self.api.event(self.on_connected)
-        self.api.event(self.on_data)
-        try:
-            await self.api.connect(reconnect=True)
-        except ApiError:
-            _LOGGER.exception("API error")
+        self._unsub_connected = self.api.listen("on_connected", self.on_connected)
+        self._unsub_data = self.api.listen("on_data", self.on_data)
+        self._unsub_connection_lost = self.api.listen(
+            "on_connection_lost", self.on_connection_lost
+        )
+
+        for _ in range(3):
+            if self.hass.is_stopping:
+                return
+
+            try:
+                await self.api.connect(reconnect=True)
+            except ApiError:
+                _LOGGER.exception("API error")
```

Comment:
Add an `else` statement to exit the loop with `return` if the connection was successful, and we can leave the loop.